### PR TITLE
Issue/1304 asyncpg connection pool

### LINKED ISSUE #1304: Add option to configure asyncpg pool
In inmanta.data.connect an asyncpg connection pool is created with default parameters. The inmanta server should have options in the database of the configuration file to set:

- minimal pool size
- maximual pool size
- connection timeout

Option to the server are added in inmanta.server.config. Use the asyncpg documentation as a basis for naming the options and documenting them (https://magicstack.github.io/asyncpg/current/api/index.html#asyncpg.pool.create_pool)

### PR DESCRIPTION
**Pull request by andraskvr**
Fixes #1304

### CODE REVIEW

File: src/inmanta/config.py [hunk 1]
```diff
@@ -158,6 +158,16 @@ def is_int(value: str) -> int:
     return int(value)
 
 
+def is_float(value: str) -> float:
```
Comment (arnaudsjs) [inline src/inmanta/config.py]: Duplicate method.

File: src/inmanta/data/__init__.py [hunk 2]
```diff
@@ -2546,9 +2546,27 @@ def set_connection_pool(pool):
...
+    database: str,
+    username: str,
+    password: str,
+    create_db_schema: bool = True,
+    connection_pool_min_size: int = 10,
+    connection_pool_max_size: int = 10,
+    database_connection_timeout=60,
 ) -> asyncpg.pool.Pool:
-    pool = await asyncpg.create_pool(host=host, port=port, database=database, user=username, password=password)
+    connection_kwargs = {"timeout": database_connection_timeout}
```
Comment (arnaudsjs) [inline src/inmanta/data/__init__.py]: Putting it as a keyword argument in the method call would improve readability. 

We should also add a test that set the config options on the Inmanta server.

File: tests/test_data.py [hunk 3]
```diff
@@ -2163,3 +2217,12 @@ def assert_match_code(code1, code2):
     res = await data.Resource.get_one(environment=res1.environment, resource_version_id=res1.resource_version_id)
 
     assert res1.attributes == res.attributes
+
+
+#
+# @pytest.mark.asyncio
```
Comment (arnaudsjs) [inline tests/test_data.py]: Why is this commented?

File: tests/test_data.py [hunk 4]
```diff
@@ -19,14 +19,68 @@
...
+
+
+@pytest.mark.asyncio
+async def test_connect_default_parameters(postgres_db, database_name: str, create_db_schema: bool = False):
+    pool: Pool = await data.connect(postgres_db.host, postgres_db.port, database_name, postgres_db.user, None, create_db_schema)
+    assert pool is not None
+    try:
+        connection: Connection = await pool.acquire()
+        assert connection is not None
+        await connection.close()
```
Comment (arnaudsjs) [inline tests/test_data.py]: connection.close() should be part of the finally block.

File: src/inmanta/server/config.py [hunk 5]
```diff
@@ -33,6 +33,13 @@
 db_name = Option("database", "name", "inmanta", "The name of the database on the postgresql server")
 db_username = Option("database", "username", "postgres", "The username to access the database in the PostgreSQL server")
 db_password = Option("database", "password", None, "The password that belong to the database user")
+db_connection_pool_min_size = Option(
```
Comment (arnaudsjs) [inline src/inmanta/server/config.py]: It seems that these connection parameter are never read by the Inmanta server to pass them towards the data.connect() method. Right?

File: src/inmanta/data/__init__.py (hunk 2 above)
Comment (wouterdb) [inline src/inmanta/data/__init__.py]: this is somewhat weird way of doing it

File: src/inmanta/data/__init__.py [hunk 6]
```diff
@@ -2546,9 +2546,27 @@ def set_connection_pool(pool):
...
+    pool = await asyncpg.create_pool(
+        host=host,
+        port=port,
+        database=database,
+        user=username,
+        password=password,
+        min_size=connection_pool_min_size,
+        max_size=connection_pool_max_size,
+        **connection_kwargs,
```
Comment (wouterdb) [inline src/inmanta/data/__init__.py]:
```suggestion
        timeout=database_connection_timeout,
```

File: src/inmanta/data/__init__.py (hunk 2 above)
Comment (wouterdb) [reply src/inmanta/data/__init__.py]: (see patch below)

File: tests/test_data.py (hunk 3 above)
Comment (wouterdb) [inline tests/test_data.py]: remove or uncomment

File: src/inmanta/data/__init__.py [hunk 7]
```diff
@@ -2546,9 +2546,26 @@ def set_connection_pool(pool):
...
-    host: str, port: int, database: str, username: str, password: str, create_db_schema: bool = True
+    host: str,
+    port: int,
+    database: str,
+    username: str,
+    password: str,
+    create_db_schema: bool = True,
+    connection_pool_min_size: int = 10,
+    connection_pool_max_size: int = 10,
+    connection_timeout=60,
```
Comment (arnaudsjs) [inline src/inmanta/data/__init__.py]:
```suggestion
    connection_timeout: float = 60,
```

Comment (andraskvr) [reply src/inmanta/data/__init__.py]: That's more precise, thank you

**Pull request merged**